stream/stream_dvbv5.c

```diff
@@ -0,0 +1,577 @@
+        {"cc", OPT_STRING(cfg_cc)},
+        {"video_pid", OPT_INT(n_vpid), M_RANGE(0, 9999)},
+        {"audio_pid", OPT_INT(n_apid), M_RANGE(0, 9999)},
+        {"verbose", OPT_INT(cfg_verbose), M_RANGE(0, 9999)},
+        {"pat", OPT_BOOL(cfg_rec_psi)},
+        {"lnb_name", OPT_STRING(cfg_lnb_name)},
+        {"lna", OPT_INT(cfg_lna), M_RANGE(-1, 1)},
+        {"sat_number", OPT_INT(cfg_sat_number), M_RANGE(0, 9999)},
+        {"diseqc_wait", OPT_INT(cfg_diseqc_wait), M_RANGE(0, 9999)},
+        {"freq_bpf", OPT_INT(cfg_freq_bpf), M_RANGE(0, 9999)},
+        {"port", OPT_INT(cfg_port), M_RANGE(0, 65536)},
+        {"server", OPT_STRING(cfg_server)},
+        {"timeout", OPT_INT(cfg_timeout), M_RANGE(0, 3600)},
+        {0}
+    },
+    .size = sizeof(struct dvbv5_opts),
+    .defaults = &(const dvbv5_opts_t){
+        .cfg_adapter = 0,
+        .cfg_frontend = 0,
+        .cfg_demux = 0,
+        .cfg_cc = NULL,
+        .n_vpid = 0,
+        .n_apid = 0,
+        .cfg_verbose = 0,
+        .cfg_rec_psi = 0,
+        .cfg_lnb_name = NULL,
+        .cfg_lna = LNA_AUTO,
+        .cfg_sat_number = -1,
+        .cfg_diseqc_wait = 0,
+        .cfg_freq_bpf = 0,
+        .cfg_port = 0,
+        .cfg_server = NULL,
+        .cfg_timeout = 120,
+    },
+};
+static int timeout_flag = 0;
+
+static void *do_timeout(void *priv_ptr) {
+    dvb_priv_t *priv = priv_ptr;
+    usleep(priv->opts->cfg_timeout * 1000000);
+    timeout_flag = 1;
+    return NULL;
+}
+
+static void dvbv5_close(stream_t *stream)
+{
+    dvb_priv_t *priv = stream->priv;
+
+    dvb_dev_free(priv->dvb);
+}
+
+static int dvbv5_stream_control(struct stream *s, int cmd, void *arg)
+{
+    MP_ERR(s, "stream control");
+    switch (cmd) {
+    case STREAM_CTRL_GET_METADATA: {
+        struct mp_tags *metadata = talloc_zero(NULL, struct mp_tags);
+        char *progname = s->path;
+        mp_tags_set_str(metadata, "title", progname);
+        *(struct mp_tags **)arg = metadata;
+        return STREAM_OK;
+    }
+    }
+    return STREAM_UNSUPPORTED;
+}
+
+static int dvbv5_streaming_read(stream_t *stream, void *buffer, int size)
+{
+    int r, pos = 0;
+
+    dvb_priv_t *priv = stream->priv;
+
+    r = dvb_dev_read(priv->dvr_fd, buffer, size);
+
+    pos += r;
+
+    return pos;
+}
+
+/*
+ * This function was adapted for MPV from v4l-utils/utils/dvb/dvbv5-zap.c
+ * version v4l-utils-1.20.0-118-g9a628fbc
+ */
+static int parse(stream_t *stream, struct dvb_v5_fe_parms *parms,
+         int *vpid, int *apid, int *sid)
+{
+    struct mpv_global *global = stream->global;
+    void *talloc_ctx;
+    char *conf_file;
+    char *channel;
+    enum dvb_file_formats input_format;
+    channel = stream->path;
+    dvb_priv_t *priv;
+    struct dvbv5_opts *args;
+
+    priv = stream->priv;
+    args = priv->opts;
+
+    input_format = dvb_parse_format("DVBV5");
+
+    struct dvb_file *dvb_file;
+    struct dvb_entry *entry;
+    int i;
+    uint32_t sys;
```

Comment:
Functions would be more readable if you declared variables only at the point where they're needed, mpv's coding style allows this.

Reply:
I like that as well but I stated in the commit/PR description I'd like to keep the source as simular as possible to dvbv5-zap.c to port stuff over.